builtin/provisioners/puppet/resource_provisioner.go

```diff
@@ -259,16 +259,29 @@ func (p *provisioner) generateAutosignToken(certname string) (string, error) {
 }
 
 func (p *provisioner) installPuppetAgentOpenSource() error {
+	task := "puppet_agent::install"
+
+	connType := p.instanceState.Ephemeral.ConnInfo["type"]
+	if connType == "" {
+		connType := "ssh"
+	}
```

Comment:
@mildwonkey I think that's probably the best way ?

Reply:
I think that makes sense, but connType was already declared so you don't use the short assignment `:=` inside the if statement:

```
	connType := p.instanceState.Ephemeral.ConnInfo["type"]
	if connType == "" {
		connType = "ssh"
	}
```

Reply:
@mildwonkey Yes fixed, that hit me when I recompiled. Try now.

Reply:
And retested on Linux using this code here https://github.com/alexharv074/terraform-puppet-provisioner-test/blob/master/main.tf#L54-L64  (I can't test this on Windows unfortunately.)